src/Cache/FileHandler.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Cache;
+
+use Symfony\Component\Filesystem\Exception\IOException;
+
+/**
+ * @author Andreas Möller <am@localheinz.com>
+ *
+ * @internal
+ */
+final class FileHandler implements HandlerInterface
+{
+    /**
+     * @var string
+     */
+    private $file;
+
+    /**
+     * @param string $file
+     */
+    public function __construct($file)
+    {
+        $this->file = $file;
+    }
+
+    public function getFile()
+    {
+        return $this->file;
+    }
+
+    public function read()
+    {
+        if (!file_exists($this->file)) {
+            return;
+        }
+
+        $content = file_get_contents($this->file);
+        $cache = @unserialize($content);
```

Comment:
@localheinz if JSON would be incorrect then assume cache file is not there.
Just like what you do now when you deserialize and get instance of other interface.
I would prefer to have invalid JSON than wrongly injected instance of not our class (as somebody could create his own class with his own code basing on our interface)